packages/content/src/attributes/types/strings/AbstractURL.ts

```diff
@@ -8,8 +8,7 @@ export abstract class AbstractURL extends AbstractString {
         min: 3,
         max: 1024,
         regExp: new RegExp(
-            // eslint-disable-next-line no-useless-escape
-            /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=\+\$,\w]+@)?[A-Za-z0-9.-]+|(?:www.|[-;:&=\+\$,\w]+@)[A-Za-z0-9.-]+)((?:\/[\+~%\/.\w-_]*)?\??(?:[-\+=&;%@.\w_]*)#?(?:[\w]*))?)/
+            /^(?:|[A-Za-z]{3,9}:(?:[/][/])?|[:](?:[/][/])?|(?:[/][/]))(?:(www[.])(?:[A-Za-z0-9\u00c4\u00e4\u00d6\u00f6\u00dc\u00fc\u00df](?:[A-Za-z\u00c4\u00e4\u00d6\u00f6\u00dc\u00fc\u00df0-9-]{0,61}[A-Za-z\u00c4\u00e4\u00d6\u00f6\u00dc\u00fc\u00df0-9]){0,1}[.])|((?!www[.])(?:[A-Za-z0-9\u00c4\u00e4\u00d6\u00f6\u00dc\u00fc\u00df](?:[A-Za-z\u00c4\u00e4\u00d6\u00f6\u00dc\u00fc\u00df0-9-]{0,61}[A-Za-z\u00c4\u00e4\u00d6\u00f6\u00dc\u00fc\u00df0-9]){0,1}[.])))[A-Za-z\u00c4\u00e4\u00d6\u00f6\u00dc\u00fc\u00df0-9](?:[A-Za-z\u00c4\u00e4\u00d6\u00f6\u00dc\u00fc\u00df0-9-]{0,61}[A-Za-z\u00c4\u00e4\u00d6\u00f6\u00dc\u00fc\u00df0-9])(?:[:][0-9]+){0,}(?:[/][A-Za-z\u00c4\u00e4\u00d6\u00f6\u00dc\u00fc\u00df0-9?#@!$&'()*+/,;=%-]{0,}){0,}$/
         )
     })
     public override value: string;
```

Comment:
1. You can replace unicode escape sequences (`\u00c4`, ...) with literal characters (`Ä`, ...) in regular expressions. This will make this madness more legible.

    ```ts
    re = /[A-Za-z0-9ÄäÖöÜüß]/
    ```

2. You're using non-capturing groups `(?:...)`, which would be useful if you actually used capture groups somewhere to _substitute_ certain parts of the string matched by the regex. But since we're just matching and not substituting here just yeet the `?:` prefix and make everything a little more legible again.

3. _www_ has no special role in a domain. It's an old convention to use the _www_ subdomain to host one's public facing website (e.g. _www.ebay.com_). It's practically meaningless nowadays, so no need to give it any separate treatment.

Also, I think you're checking that there's no more than one _www_ subdomain in the url. This is perfectly legal, however. `www.foo.www.www.enmeshed.eu` is a perfectly fine domain. 

4. `{0,1}` is better expressed as `?` in a regex, e.g. `([a-z]{0,61})?` instead of `([a-z]{0,61}){0,1}`.

5. I assume `(\:[0-9]+)` is supposed to check for the port of a host. What's the leading backslash good for? I don't think you need it.

6. `{0,}` is better expressed as `*` in a regex, e.g. `([a-z]{0,61})*` instead of `([a-z]{0,61}){0,}`.

7. Is `[.]` supposed to match a literal period character '.'? Use the standard `\.` (a `.` inside a regex means "anything but a newline", hence you have to escape it).

8. Can you help me make sense of those segments of up to 61 characters? I don't quite understand what these are about.

Reply:
1. & 2. Thank you for the hints!
3. That wasn't my intension, I didn't want that you can write www.google for example, but for other reasons www.foo.www.www.enmeshed.eu didn't work because I overlooked it, but now it works when I will push the commit. 
4.&6. I know but I thought that someone who don't know what + and * means would understand it better but anyway it's shorter so I change it. 
5.&7. I used \: for escaping it but when you put it in []-brackets you don't need to escape and therefore in 7. I also put the . into [] 
By the way If you don't avoid that typescript deletes unnecessary escape characters you will have the problem that were there in further versions. 
8. Between the points are the domains and every domain can't have more than 63 characters and can't end or begin with a -